### DISCUSSION TITLE
useFetch vs useAsyncData for api layer

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/nuxt#25400] [Questions] [answered]
The docs on data fetching state that:
> There are some cases when using the [useFetch](https://nuxt.com/docs/api/composables/use-fetch) composable is not appropriate, for example when a CMS or a third-party provide their own query layer. In this case, you can use [useAsyncData](https://nuxt.com/docs/api/composables/use-async-data) to wrap your calls and still keep the benefits provided by the composable.

Why is `useFetch` not appropriate if all it is, is a wrapper around `useAsyncData` and `$fetch`?

**Comment by TheAlexLichter** [comment, marked as answer]
It is appropriate, but when you e.g. have to use a different fetching library, like Nuxt Content's `queryContent`, you can't use `useFetch` for that 😋

**Reply by dbzx10299** [reply]
Aha got it, because `useFetch` does not accept a callback
